doc/eclipse.html

```diff
@@ -335,6 +343,52 @@ <h2 align="center">TestNG Eclipse plug-in</h2>
+
+TestNG M2E Integration plugin was introduced since 6.9.10 as an optional plugin.
+It allows you to run your tests with System Properties or JVM settings, which are defined by maven-surefire-plugin or maven-failsafe-plugin of pom.xml, to be appended to the runtime TestNG process.
+
+Once this plugin installed, you can see a dedicate preference page (workspace level settings):
+<p align="center">
+<img border="0" src="http://testng.org/pictures/m2e-pref.png" />
+</p>
+or, project properties page (as name implied, can have project specific settings to override the workspace ones:
+<p align="center">
+<img border="0" src="http://testng.org/pictures/m2e-prop.png" />
+</p>
+
+Let's say there is maven-surefire-plugin confguration in your pom.xml:
+<pre class="brush: xml">
+        &lt;artifactId&gt;maven-surefire-plugin&lt;/artifactId&gt;
+        &lt;configuration&gt;
+          &lt;suiteXmlFiles&gt;
+            &lt;suiteXmlFile&gt;test-suite/testng.xml&lt;/suiteXmlFile&gt;
+          &lt;/suiteXmlFiles&gt;
+          &lt;argLine&gt;-javaagent:${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar
+                        -Xmx1024m -XX:MaxPermSize=512m -Xms256m -Xmx1024m -XX:PermSize=128m 
+                        -Dversion=${project.version}
+          &lt;/argLine&gt;
+          &lt;systemPropertyVariables&gt;
+            &lt;foo&gt;${foo.value}&lt;/foo&gt;
+          &lt;/systemPropertyVariables&gt;
+          &lt;environmentVariables&gt;
+            &lt;bar&gt;${bar.value}&lt;/bar&gt;
+          &lt;/environmentVariables&gt;
+        &lt;/configuration&gt;
+</pre>
+
+you can pass the following Maven configuraion parameters to TestNG process:
+<ul>
+<li><b><a href="http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#argLine">argLine</a></b>: the JVM arguments</li>
+<li><b><a href="http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#systemPropertyVariables">systemPropertyVariables</a></b>: the system properties</li>
+<li><b><a href="http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#environmentVariables">environmentVariables</a></b>: the environment variables</li>
+</ul>
+<p>
+NOTE: as the snippet maven-surefire-plugin confguration above, the properties placeholder (e.g., ${aspectj.version}) in `argLine` or `systemPropertyVariables`, etc. will be substituted and correctly passed to TestNG process as lone as the properties are visible on the maven project (for example, defined in current pom.xml, or inherited from parent pom.xml, etc.)<br/>
+NOTE: if youre maven-surefire-plugin is defined in maven profile, to make TestNG M2E Integration plugin works as expected, you need to select the maven profile which contains the maven-surefire-plugin configuration: "Rigth click on the project -> Maven -> Select Maven Profiles...", then check the profiles as you like, if multi profiles selected, the first one has maven-surefire-plugin wins.
+</p>
+
```

Comment:
(Note: I surrounded identifiers with &lt;code&gt; tags but github is interpreting them...)